### PR TITLE
Add compton user service

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

This adds a NixOS module for compton so that it can be configured declaratively
and run as a user service.
###### Things done
- [x] Tested using sandboxing
- [x] Tested execution of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/compton.nix [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        unredir-if-possible = true;
+        dbe = true;
+      '';
+      description = ''
+        Additional Compton configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
```
Comment (RamKromberg) [inline nixos/modules/services/x11/compton.nix]: You can add `environment.systemPackages = [ pkgs.compton ];` here and automate compton's install when enabled.

File: nixos/modules/services/x11/compton.nix [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      example = 60;
+      description = ''
+       Screen refresh rate (0 = automatically detect).
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.compton;
+      defaultText = "pkgs.compton";
```
Comment (RamKromberg) [inline nixos/modules/services/x11/compton.nix]: `example = literalExample "pkgs.compton";` might be useful. 

In the manual's appendix (https://nixos.org/nixos/manual/options.html) you can see how _boot.kernelPackages_ uses it.

File: nixos/modules/services/x11/compton.nix (hunk 1 above)
Comment (rnhmjoj) [inline nixos/modules/services/x11/compton.nix]: Sure, I forgot about this.

File: nixos/modules/services/x11/compton.nix (hunk 2 above)
Comment (rnhmjoj) [inline nixos/modules/services/x11/compton.nix]: Thanks

**Comment by rnhmjoj** [comment]
Can it be merged now?

**Pull request merged**